std/experimental/color/package.d

```diff
@@ -0,0 +1,474 @@
+{
+    // cast along a conversion path to reach our target conversion
+    alias Path = ConversionPath!(From, To);
+
+    // no conversion is necessary
+    static if(Path.length == 0)
+        return color;
+    else static if(Path.length > 1)
+    {
+        // we need to recurse to trace a path via the first common ancestor
+        static if(__traits(compiles, From.convertColorImpl!(Path[0])(color)))
+            return convertColor!To(From.convertColorImpl!(Path[0])(color));
+        else
+            return convertColor!To(To.convertColorImpl!(Path[0])(color));
+    }
+    else
+    {
+        static if(__traits(compiles, From.convertColorImpl!(Path[0])(color)))
+            return From.convertColorImpl!(Path[0])(color);
+        else
+            return To.convertColorImpl!(Path[0])(color);
+    }
+}
+///
+unittest
+{
+    assert(convertColor!(RGBA8)(convertColor!(XYZ!float)(RGBA8(0xFF, 0xFF, 0xFF, 0xFF))) == RGBA8(0xFF, 0xFF, 0xFF, 0));
+}
+
+
+/**
+* Create a color from hex strings in the standard forms: (#/$/0x)rgb/argb/rrggbb/aarrggbb
+*/
+Color colorFromString(Color = RGB8, C)(const(C)[] hex) if(isSomeChar!C)
+{
+    static ubyte val(C c)
+    {
+        if(c >= '0' && c <= '9')
+            return cast(ubyte)(c - '0');
+        else if(c >= 'a' && c <= 'f')
+            return cast(ubyte)(c - 'a' + 10);
+        else if(c >= 'A' && c <= 'F')
+            return cast(ubyte)(c - 'A' + 10);
+        else
+            assert(false, "Invalid hex string");
+    }
+
+    if(hex.length > 0 && (hex[0] == '#' || hex[0] == '$'))
+        hex = hex[1..$];
+    else if(hex.length > 1 && (hex[0] == '0' && hex[1] == 'x'))
+        hex = hex[2..$];
+
+    if(hex.length == 3)
+    {
+        ubyte r = val(hex[0]);
+        ubyte g = val(hex[1]);
+        ubyte b = val(hex[2]);
+        return cast(Color)RGB8(cast(ubyte)(r | (r << 4)), cast(ubyte)(g | (g << 4)), cast(ubyte)(b | (b << 4)));
+    }
+    if(hex.length == 4)
+    {
+        ubyte a = val(hex[0]);
+        ubyte r = val(hex[1]);
+        ubyte g = val(hex[2]);
+        ubyte b = val(hex[3]);
+        return cast(Color)RGBA8(cast(ubyte)(r | (r << 4)), cast(ubyte)(g | (g << 4)), cast(ubyte)(b | (b << 4)), cast(ubyte)(a | (a << 4)));
+    }
+    if(hex.length == 6)
+    {
+        ubyte r = cast(ubyte)(val(hex[0]) << 4) | val(hex[1]);
+        ubyte g = cast(ubyte)(val(hex[2]) << 4) | val(hex[3]);
+        ubyte b = cast(ubyte)(val(hex[4]) << 4) | val(hex[5]);
+        return cast(Color)RGB8(r, g, b);
+    }
+    if(hex.length == 8)
+    {
+        ubyte a = cast(ubyte)(val(hex[0]) << 4) | val(hex[1]);
+        ubyte r = cast(ubyte)(val(hex[2]) << 4) | val(hex[3]);
+        ubyte g = cast(ubyte)(val(hex[4]) << 4) | val(hex[5]);
+        ubyte b = cast(ubyte)(val(hex[6]) << 4) | val(hex[7]);
+        return cast(Color)RGBA8(r, g, b, a);
+    }
+    else
+    {
+        // TODO: should we look up colors from the W3C color table by name?
+
+        assert(false, "Invalid hex string!");
```

Comment:
I guess I should throw? That's a hassle, because I have `nothrow` at the top of the file! >_<
I either need to rearrange the file (which sucks) or put `nothrow` on each function instead?

Also, what exception should I throw for invalid input? I'm not a fan of exceptions :/